File: Back/Gestion des conges/src/main/java/com/example/gestion_des_conges/Entities/Employee.java
package com.example.gestion_des_conges.Entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;
import lombok.experimental.FieldDefaults;

import javax.persistence.*;
import java.io.Serializable;
import java.util.List;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class Employee implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Integer idEmp;

    String email;

    String username;

    String motDePass;

    String nom;

    String prenom;

    String adresse;

    double soldeConge;

    boolean autorisationJour = false;

    long autorisationDuration = 6;

    @Enumerated(EnumType.STRING)
    Statut statut;

    String entite;

    @ManyToOne
    Role role;

    @ManyToOne
    @JsonIgnore
    Employee superviseur;

    @OneToMany(mappedBy = "demandeur")
    @JsonIgnore
    List<Conge> listDemande;

    @OneToMany(mappedBy = "validateur")
    @JsonIgnore
    List<Conge> listValidation;
    @ManyToOne
    Politique politique;

    @OneToMany(mappedBy = "demandeur")
            @JsonIgnore
    List<Autorisation> autorisations;
}

File: Front/Gestion_Conge/src/app/Model/Employee.ts
import { Conge } from "./Conge";
import { Politique } from "./Politique";
import { Role } from "./Role";

export class Employee {
    idEmp?: string;
    email?: string;
    username?: string;
    motDePass?: string;
    nom?: string;
    prenom?: string;
    adresse?: string;
    soldeConge?: string;
    statut?: string;
    entite?: string;
    role?: Role;
    superviseur?: Employee;
    listDemande?: Conge[];
    listValidation?: Conge[];
    politique?: Politique;
  }